src/main/java/com/google/sps/app/servlets/LogIn.java

```diff
@@ -0,0 +1,61 @@
+package com.google.sps.app.servlets;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.api.gax.rpc.InvalidArgumentException;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.gson.Gson;
+import com.google.sps.util.Validator;
+
+@WebServlet("/LogIn")
+public class LogIn extends HttpServlet {
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String username = request.getParameter("text-input-user-name");
+        if(!Validator.validUserName(username)) {
+            response.getWriter().println("Invalid Username");
+            return;
+        }
+
+        UUID userId;
+        try {
+            userId = readFromDatastore(username);
+        } catch (IllegalArgumentException e) {
+            response.getWriter().println(e.getMessage());
+            return;
+        }
+
+        Gson gson = new Gson();
+
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(userId));
+    }
+
+    public UUID readFromDatastore(String username) throws IllegalArgumentException {
+
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+        Query<Entity> query = 
+            Query.newEntityQueryBuilder().setKind("User").build();
+        QueryResults<Entity> results = datastore.run(query);
+        
+        while (results.hasNext()) {
```

Comment:
You can add filters to queries to only fetch the username you want, like below. That is the right way to do it.
https://cloud.google.com/datastore/docs/concepts/queries#datastore-datastore-basic-query-java

Query.newEntityQueryBuilder()
        .setKind("User")
        .setFilter(PropertyFilter.eq(username, true))
        .setOrderBy(OrderBy.desc("priority"))
        .build();

Just FYI: You may get multiple matches, since you are currently not enforcing name uniqueness. That should go away once you use actual login. But you will still have to see if the user has already signed up.

Reply:
I've used GQL queries to get matching results. I'll get to work on verifying unique names too.